dvc_gs/tests/fixtures.py

```diff
@@ -24,9 +19,21 @@ def gs_creds(tmp_path_factory):
 
 
 @pytest.fixture
-def make_gs(gs_creds):
-    def _make_gs():
-        return GCP(GCP.get_url(), gs_creds)
+def make_gs(request):
+    if os.getenv("DVC_GS_NO_EMULATOR", None):
+
+        def _make_gs():
+            return GCP(GCP.get_url(), "")
+
+    else:
+        gcs_path = request.getfixturevalue("gcs_path")
+        fake_gcs_server = request.getfixturevalue("fake_gcs_server")
+
+        def _make_gs():
+            return FakeGCP(
+                str(gcs_path).replace("gcs://", "gs://"),
+                endpoint_url=fake_gcs_server,
+            )
 
     return _make_gs
```

Comment:
Let's do similar to `dvc-s3`: have `gs` fixture that is always using fake server and have a `real_gs` fixture that uses real gs. Tests should use `gs`(fake) always, but if one would need to debug stuff, he could use `real_gs` manually.